src/validation.cpp

```diff
@@ -726,9 +715,42 @@ class MemPoolAccept
 
     Chainstate& m_active_chainstate;
 
+    // Fields below are per *sub*package state and must be reset prior to subsequent
+    // AcceptSingleTransaction and AcceptMultipleTransactions invocations
+
+    /** Aggregated modified fees of all transactions, used to calculate package feerate. */
+    CAmount m_total_modified_fees{0};
+    /** Aggregated virtual size of all transactions, used to calculate package feerate. */
+    int64_t m_total_vsize{0};
+
+    // RBF-related members
     /** Whether the transaction(s) would replace any mempool transactions and/or evict any siblings.
      * If so, RBF rules apply. */
     bool m_rbf{false};
+    /** All directly conflicting mempool transactions and their descendants. */
+    CTxMemPool::setEntries m_all_conflicts;
+    /** Mempool transactions that were replaced. */
+    std::list<CTransactionRef> m_replaced_transactions;
+
+    /** Total modified fees of mempool transactions being replaced. */
+    CAmount m_conflicting_fees{0};
+    /** Total size (in virtual bytes) of mempool transactions being replaced. */
+    size_t m_conflicting_size{0};
+
+    /** Re-set sub-package state to not leak between evaluations */
+    void ClearSubPackageState() EXCLUSIVE_LOCKS_REQUIRED(cs_main, m_pool.cs)
```

Comment:
Do you think it might make sense to explicitly group these variables into a single struct that makes it clear we have one grouping of per-subpackage state?  All these variables that are at the same scope level as other things that don't get reset seems maybe a bit messy...

Reply:
Probably worth doing now to make it more clear over time. Let me know what you think.

Reply:
Seems more readable to me, thanks.

Reply:
Looks like the commit message for commit 6b3373248656ef45d73262aa6dba3aec5866f7ca should be updated now, sorry!

Reply:
touched up the commit message